src/tests/system/tests/test_passkey.py

```diff
@@ -78,30 +84,352 @@ def test_passkey__register__ipa(ipa: IPA, moduledatadir: str, testdatadir: str):
+        3. Bring SSSD offline explicitly.
+        4. Check su offline authentication of the user.
+    :expectedresults:
+        1. User su authenticated successfully.
+        2. Firewall rules dropped.
+        3. SSSD is offline.
+        4. Offline su authentication is successful.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+
+    client.sssd.domain["cache_credentials"] = "True"
+    client.sssd.domain["krb5_store_password_if_offline"] = "True"
+    client.sssd.pam["offline_credentials_expiration"] = "0"
+
+    client.sssd.start()
+
+    # First time check authentication to cache the user
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+    # Block KDC, LDAP and Global Catalog ports.
+    provider.firewall.drop([88, 389, 3268])
+
+    # There might be active connections that are not terminated by creating firewall rule.
+    # We need to terminated it by bringing SSSD to offline state explicitly.
+    client.sssd.bring_offline()
+
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__user_fetch_from_cache(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Fetch a user from cache for LDAP, IPA, AD and Samba server
+    :setup:
+        1. Add a user in LDAP, IPA, AD and Samba with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check a user lookup.
+        2. Check a user from cache using ldbsearch command.
+    :expectedresults:
+        1. A user looked up successfully.
+        2. Successfully get the user from ldbsearch command.
+    :customerscenario: False
+    """
+
+    suffix = type(provider).__name__.lower()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    result = client.tools.id("user1")
+    # ldb_search = 'ldbsearch -H /var/lib/sss/db/cache_test.ldb -b "cn=test,cn=sysdb"'
+    # output = client.host.ssh.run(ldb_search, raise_on_error=False)
```

Comment:
Drop the comments if the new function call is working.

Reply:
yes, updated.